tsdb/head_append.go

```diff
@@ -312,24 +314,73 @@ type exemplarWithSeriesRef struct {
 	exemplar exemplar.Exemplar
 }
 
-type headAppender struct {
-	head          *Head
-	minValidTime  int64 // No samples below this timestamp are allowed.
-	mint, maxt    int64
-	headMaxt      int64 // We track it here to not take the lock for every sample appended.
-	oooTimeWindow int64 // Use the same for the entire append, and don't load the atomic for each sample.
+// appendType describes sample types we need to distinguish for append batching.
+// We need separate types for everything that goes into a different WAL record
+// type or into a different chunk encoding.
+type sampleType byte
```

Comment:
Hm, naming of this can be confusing vs metric type. I guess `sample` prefix is helping, but maybe `sampleEncoding`, `sampleStructure` is better?

Is this an TSDB appender internal dimension or is this grouping an aftermath of our WAL record/chunk encoding decisions. As-is - if we ever decide to have a SINGLE record that combines all native histograms, then separate for floats (just 2), same for TSDB chunks... would that change this enum to float vs histogram?

Is type and unit label helping in any form? I think it helps for stFloat vs stHistogram&co, reducing the problem space significantly, no?

Reply:
These `sampleType`s are really internal to the headAppender. They are informed by what is treated differently downstream (in TSDB or in PRW, respectively, and funny enough, these are different things, and we need to do the Cartesian product of those).

Reply:
As this is an internal concept, I don't see the naming as too critical. If you have a better proposal, I'm all ears.